packages/form-layout/src/vaadin-form-item.js

```diff
@@ -159,17 +160,62 @@ class FormItem extends ThemableMixin(PolymerElement) {
     return 'vaadin-form-item';
   }
 
+  constructor() {
+    super();
+    this.__updateInvalidState = this.__updateInvalidState.bind(this);
+    this.__contentFieldObserver = new MutationObserver(() => this.__updateRequiredState(this.__contentField.required));
+  }
+
   /** @private */
   _onLabelClick() {
-    const firstContentElementChild = Array.prototype.find.call(
-      this.$.contentSlot.assignedNodes(),
-      (e) => e.nodeType === Node.ELEMENT_NODE
-    );
+    const firstContentElementChild = this.__getContentChildNodes()[0];
     if (firstContentElementChild) {
       firstContentElementChild.focus();
       firstContentElementChild.click();
     }
   }
+
+  /** @private */
+  __getContentChildNodes() {
+    return [...this.$.contentSlot.assignedNodes()].filter((e) => e.nodeType === Node.ELEMENT_NODE);
+  }
+
+  /** @private */
+  __onContentSlotChange() {
+    if (this.__contentField) {
+      // Discard the old field
+      this.__updateRequiredState(false);
+      this.__contentFieldObserver.disconnect();
+      delete this.__contentField;
+    }
+
+    const contentFields = this.__getContentChildNodes().filter((node) => 'checkValidity' in node);
```

Comment:
Let's consider using the same check as in `vaadin-custom-field` for consistency:
https://github.com/vaadin/web-components/blob/0cbdfdbe4c93a57f1554e426de93ad5788aa6b4e/packages/custom-field/src/vaadin-custom-field.js#L271

BTW, this is not the case anymore, but in the past e.g. `vaadin-checkbox-group` only had `validate()`.

Reply:
Done.